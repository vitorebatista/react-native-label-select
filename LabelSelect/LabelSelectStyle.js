/**
 * Created by TinySymphony on 2017-01-03.
 */

import {
  Dimensions,
  StyleSheet
} from 'react-native';

const window = Dimensions.get('window');
const { width, height, scale } = window;
const borderScale = 2 / scale;

export const Color = {
  disableColor: '#eaeaea',
  main: '#40cca2'
};

export default StyleSheet.create({
  selectedView: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexWrap: 'wrap'
  },
  selectedItem: {
    margin: 4,
    borderWidth: borderScale,
    borderRadius: 6,
    borderColor: '#aaa',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    overflow: 'hidden',
    backgroundColor: '#f6f6f6'
  },
  addItem: {
    padding: 0
  },
  disableColor: {
    backgroundColor: Color.disableColor
  },
  labelText: {
    padding: 6,
    fontSize: 14,
    lineHeight: 14,
    maxWidth: 300
  },
  closeContainer: {
    padding: 8,
    borderLeftWidth: 2 / scale,
    borderLeftColor: '#c8c8c8'
  },
  closeIcon: {
    width: 10,
    height: 10
  },
  viewClose: {
    margin: -1.3
  },
  textClose: {
    fontFamily: 'Arial',
    color: '#999',
    fontSize: 12,
    marginLeft: 2,
    marginRight: 2
  },
  addIcon: {
    width: 12,
    height: 12
  },
  addText: {
    color: '#999',
    fontSize: 18
  },
  viewAddText: {
    justifyContent: 'center',
    alignContent: 'center',
    flexDirection: 'row',
    height: 26,
    width: 26
  },
  modalMask: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#00000077'
  },
  modalContainer: {},
  modal: {
    height: height * 0.6,
    width: width * 0.6,
    overflow: 'hidden',
    borderRadius: 10,
    backgroundColor: '#fff'
  },
  title: {
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderBottomWidth: 2 / scale,
    borderBottomColor: '#bbb'
  },
  titleText: {
    fontSize: 18,
    lineHeight: 20
  },
  scrollView: {
    flex: 1
  },
  buttonView: {
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  modalButton: {
    height: 40,
    width: width * 0.3,
    paddingLeft: 20,
    paddingRight: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Color.main
  },
  modalItem: {
    height: 50,
    paddingHorizontal: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 2 / scale,
    borderBottomColor: '#bbb'
  },
  modalText: {
    fontSize: 16,
    width: width * 0.6 - 70
  },
  buttonText: {
    color: '#fff',
    fontSize: 16
  },
  confirmButton: {
    borderLeftWidth: borderScale,
    borderLeftColor: '#fff'
  },
  outerCircle: {
    borderWidth: borderScale,
    borderColor: '#888',
    width: 20,
    height: 20,
    borderRadius: 10,
    overflow: 'hidden',
    justifyContent: 'center',
    alignItems: 'center'
  },
  enableCircle: {
    borderColor: Color.main
  },
  innerCircle: {
    backgroundColor: Color.main,
    width: 16,
    height: 16,
    borderRadius: 8,
    overflow: 'hidden'
  },
  disableText: {
    color: '#999'
  }
});
