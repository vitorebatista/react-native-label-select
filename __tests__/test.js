import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import LabelSelect from '../LabelSelect';

import mock from './__mock__/mock';

const {Label, ModalItem} = LabelSelect;

let selectedItems = mock.selectedList.map((item, index) =>
  <Label
    key={'label-' + index}
    data={item}
    onCancel={() => {mock.selectedList.splice(index);}}>
    {item.text}
  </Label>
);

let otherItems = mock.list.map((item, index) =>
  <ModalItem
    key={'modal-item-' + index}
    data={item}>
    {item.text}
  </ModalItem>
);

// snapshot test

it('renders enabled LabelSelect', () => {
  let tree = renderer.create(
    <LabelSelect
      title="Test1"
      onConfirm={() => {}}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders readOnly LabelSelect', () => {
  let tree = renderer.create(
    <LabelSelect
      readOnly={true}
      title="Test2"
      onConfirm={() => {}}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders disabled LabelSelect', () => {
  let tree = renderer.create(
    <LabelSelect
      enable={false}
      title="Test3"
      onConfirm={() => {}}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a modal item', () => {
  let item = mock.list[0];
  expect(renderer.create(
    <ModalItem data={item}>{item.text}</ModalItem>
  )).toMatchSnapshot();
});

// enzyme test

it('interact with modal', () => {
  let arr = [];
  const item = mock.list[0];
  const tree = shallow(
    <LabelSelect
      title="Enzyme Test"
      onConfirm={(list) => {arr = list;}}
      enableAddBtn={true}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  );
  let select = tree.instance();
  expect(tree.find('TouchableHighlight').length).toEqual(4);
  select.openModal();
  expect(tree.state('isModalVisible')).toEqual(true);
  select.toggleSelect(item);
  select.confirmSelect();
  expect(arr[0]).toEqual(item);
  expect(tree.state('isModalVisible')).toEqual(false);
  select.openModal();
  expect(tree.state('isModalVisible')).toEqual(true);
  select.cancelSelect();
  expect(tree.state('isModalVisible')).toEqual(false);
  select.confirmSelect();
});

it('selecte a item', () => {
  let data = mock.list[0];
  let result;
  const tree = shallow(
    <ModalItem
      data={data}
      toggleSelect={(item) => {result = item;}}
      >
      {data.text}
    </ModalItem>
  );
  let instance = tree.instance();
  instance._toggleSelect();
  expect(result).toEqual(data);
});
