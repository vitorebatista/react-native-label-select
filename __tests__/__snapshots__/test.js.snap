exports[`test renders a modal item 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsisMode="tail"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 16,
            "width": 380,
          },
          Object {},
        ]
      }>
      C
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#888",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 20,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": 20,
          },
          Object {},
          Object {},
        ]
      } />
  </View>
</View>
`;

exports[`test renders disabled LabelSelect 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Object {},
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#eaeaea",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsisMode="tail"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "maxWidth": 300,
            "padding": 6,
          },
          Object {
            "color": "#999",
          },
          Object {},
        ]
      }>
      A
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#eaeaea",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsisMode="tail"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "maxWidth": 300,
            "padding": 6,
          },
          Object {
            "color": "#999",
          },
          Object {},
        ]
      }>
      B
    </Text>
  </View>
  <Modal
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <View
          style={Object {}}>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "height": 800.4,
                  "overflow": "hidden",
                  "width": 450,
                },
                Object {},
              ]
            }>
            <View
              style={
                Object {
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 18,
                    "lineHeight": 20,
                  }
                }>
                Test3
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 720.4,
                }
              }>
              <ScrollView />
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 10,
                    "borderBottomRightRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  },
                  Object {},
                ]
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#40cca2",
                        "height": 40,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {},
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                        },
                        Object {},
                      ]
                    }>
                    Cancel
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#40cca2",
                        "height": 40,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {
                        "borderLeftColor": "#fff",
                        "borderLeftWidth": 1,
                      },
                      Object {},
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                        },
                        Object {},
                      ]
                    }>
                    Confirm
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`test renders enabled LabelSelect 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Object {},
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsisMode="tail"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "maxWidth": 300,
            "padding": 6,
          },
          false,
          Object {},
        ]
      }>
      A
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderLeftColor": "#c8c8c8",
            "borderLeftWidth": 1,
            "padding": 8,
          },
        ]
      }
      testID={undefined}>
      <View>
        <Image
          resizeMode="cover"
          source={
            Object {
              "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAABnElEQVRYR8WXy1HDMBCG/734Sjoh6QAqgBKgg/gij29wtC4uAegAKoASXEJKCFfPeMTIYxn5IVmvTHJM4v0+rbS7MuHKH7oyH6NAXde7tm3fpBARvTLGmpRynPM9gD4+gGcVfxTgnH8CeBj+cAZwn0pigH8D2A3xG8bYoV+sWuVMQH6dRGIFLmN/McYe5wJ7IcQPEd1oqY+SWIMLIX6J6G6xBRIqH0gl4QKfZEDbimgJV/iqQGwmfOBGgVAJX7hVwFciBL4p4CoRCncS2JIYDq/eZDAvNVtHdZ4FlhKV8VWH84I7Z2CjRMcF+qxcPeScgS2JELh3BtR5ADDZ85gB5pUBw2DRz5j37HAWMJVan8aIAeYkYKtzKRAzwDYFXJpMzBS1CrjAY6eoUcAHHiOxKhACD5VYCMTAQyQmAingvhL6rVje24On2trEM1VH13WHsixPk1ZcVVVDRLcqUGhvn4sYJJbXcl0gFdy0HUKIj6IoniYZGEzfiUj282OqtyJdAsCLEOKcZdkxz3PJ+X8zst1aLvnbZiu+JFzG/gPiB7Awgm9hrgAAAABJRU5ErkJggg==",
            }
          }
          style={
            Object {
              "height": 10,
              "width": 10,
            }
          } />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsisMode="tail"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "maxWidth": 300,
            "padding": 6,
          },
          false,
          Object {},
        ]
      }>
      B
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderLeftColor": "#c8c8c8",
            "borderLeftWidth": 1,
            "padding": 8,
          },
        ]
      }
      testID={undefined}>
      <View>
        <Image
          resizeMode="cover"
          source={
            Object {
              "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAABnElEQVRYR8WXy1HDMBCG/734Sjoh6QAqgBKgg/gij29wtC4uAegAKoASXEJKCFfPeMTIYxn5IVmvTHJM4v0+rbS7MuHKH7oyH6NAXde7tm3fpBARvTLGmpRynPM9gD4+gGcVfxTgnH8CeBj+cAZwn0pigH8D2A3xG8bYoV+sWuVMQH6dRGIFLmN/McYe5wJ7IcQPEd1oqY+SWIMLIX6J6G6xBRIqH0gl4QKfZEDbimgJV/iqQGwmfOBGgVAJX7hVwFciBL4p4CoRCncS2JIYDq/eZDAvNVtHdZ4FlhKV8VWH84I7Z2CjRMcF+qxcPeScgS2JELh3BtR5ADDZ85gB5pUBw2DRz5j37HAWMJVan8aIAeYkYKtzKRAzwDYFXJpMzBS1CrjAY6eoUcAHHiOxKhACD5VYCMTAQyQmAingvhL6rVje24On2trEM1VH13WHsixPk1ZcVVVDRLcqUGhvn4sYJJbXcl0gFdy0HUKIj6IoniYZGEzfiUj282OqtyJdAsCLEOKcZdkxz3PJ+X8zst1aLvnbZiu+JFzG/gPiB7Awgm9hrgAAAABJRU5ErkJggg==",
            }
          }
          style={
            Object {
              "height": 10,
              "width": 10,
            }
          } />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#f6f6f6",
            "borderColor": "#aaa",
            "borderRadius": 6,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 4,
            "overflow": "hidden",
          },
          Object {
            "padding": 7,
          },
        ],
      ]
    }
    testID={undefined}>
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAA7ElEQVRYR+2X0QnCQAyG87cLuIF1EusEOoq+NPTJ+lSuLzqKTmDdRDdwgbvISRVEhGuVFiGBe8td/nyB+wlo4MDA9amzgLIskyiKjr4B59wsz/Nzl2Y6CzDGLAFsfVERWTHzrm8BBYB1I2DDzIUKUAJKQAn8NwH/t8dxPAcwCulERFIi8sdHDaAOvHe11h4e3vH0gqqqvJmMQx75NkdEzsw88e8MIoCILlmWJS8CGntdEFHQCAB4/NOGxklEgkZARFfn3P5tBG2xGmPUDZWAElACSmDYxaSx7/v/b61Ne1/N2nrHp/zOu+GvBNwAa6vsIVXzFTsAAAAASUVORK5CYII=",
        }
      }
      style={
        Object {
          "height": 12,
          "width": 12,
        }
      } />
  </View>
  <Modal
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <View
          style={Object {}}>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "height": 800.4,
                  "overflow": "hidden",
                  "width": 450,
                },
                Object {},
              ]
            }>
            <View
              style={
                Object {
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 18,
                    "lineHeight": 20,
                  }
                }>
                Test1
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 720.4,
                }
              }>
              <ScrollView />
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 10,
                    "borderBottomRightRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  },
                  Object {},
                ]
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#40cca2",
                        "height": 40,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {},
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                        },
                        Object {},
                      ]
                    }>
                    Cancel
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#40cca2",
                        "height": 40,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {
                        "borderLeftColor": "#fff",
                        "borderLeftWidth": 1,
                      },
                      Object {},
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                        },
                        Object {},
                      ]
                    }>
                    Confirm
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`test renders readOnly LabelSelect 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Object {},
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsisMode="tail"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "maxWidth": 300,
            "padding": 6,
          },
          false,
          Object {},
        ]
      }>
      A
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsisMode="tail"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "maxWidth": 300,
            "padding": 6,
          },
          false,
          Object {},
        ]
      }>
      B
    </Text>
  </View>
  <Modal
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <View
          style={Object {}}>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 10,
                  "height": 800.4,
                  "overflow": "hidden",
                  "width": 450,
                },
                Object {},
              ]
            }>
            <View
              style={
                Object {
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 18,
                    "lineHeight": 20,
                  }
                }>
                Test2
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 720.4,
                }
              }>
              <ScrollView />
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 10,
                    "borderBottomRightRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  },
                  Object {},
                ]
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#40cca2",
                        "height": 40,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {},
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                        },
                        Object {},
                      ]
                    }>
                    Cancel
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#40cca2",
                        "height": 40,
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {
                        "borderLeftColor": "#fff",
                        "borderLeftWidth": 1,
                      },
                      Object {},
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                          "fontSize": 16,
                        },
                        Object {},
                      ]
                    }>
                    Confirm
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
